meinberlin/apps/plans/views.py

```diff
@@ -97,6 +100,25 @@ def get_participation_choices(self):
         choices = [str(choice[1]) for choice in Plan.participation.field.choices]
         return json.dumps(choices)
 
+    def get_search_profile(self):
+        if (
+            self.request.GET.get("search-profile", None)
+            and self.request.user.is_authenticated
+        ):
+            search_profile_id = self.request.GET.get("search-profile")
+            try:
+                search_profile = SearchProfile.objects.get(
+                    id=search_profile_id, user=self.request.user
+                )
+                return (
+                    JSONRenderer()
+                    .render(SearchProfileSerializer(instance=search_profile).data)
+                    .decode("utf-8")
+                )
+            except SearchProfile.DoesNotExist:
+                pass
+        return None
```

Comment:
@vellip How do we know that one search profile is coming through? Since a user can have several. I haven't followed the fronted design for the plans.

Reply:
Nevermind, I got it now. You are choosing one, to pass its filters